muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/discussion/index.ts

```diff
@@ -716,15 +717,16 @@ const createDiscussionThread: CreateDiscussionThreadTriggerType =
           )()
         );
 
+        const hash = window.location.hash.replace("#", "").split("/");
+
+        const areaId = discussion.areaId ? discussion.areaId : 0;
+
+        const areaString = hash.includes("subs")
+          ? "subs"
+          : areaId + "/" + discussion.page;
+
         window.location.hash =
-          newThread.forumAreaId +
-          "/" +
-          (newThread.forumAreaId === discussion.areaId
-            ? discussion.page
-            : "1") +
-          "/" +
-          newThread.id +
-          "/1";
+          areaString + "/" + newThread.forumAreaId + "/" + newThread.id + "/1";
 
```

Comment:
Add little bit documentation what for and how hash is build. Especially for areaString.